### PR TITLE
VerifReg: Resolve Verifier and Client Addresses

### LINKED ISSUE #556: Verified Registry needs to resolve addresses
The Verified Registry currently only works if the `Address` provided in `AddVerifierParams` is always an ID-address. If a pubkey address is sent, that address won't actually be able to add verified clients, since it won't be found in `GetVerifier`, which will be called with its ID address (since `rt.Message().Caller()` always returns an ID address).

Assuming this isn't okay, we need to do similar address resolution as the multisig's `resolve()`. It'll be a little trickier in this case, since we probably want to overwrite pubkey addresses with ID addresses in `st.Verifiers` when they first become available lest we run the risk of having two different entries in the map for the same "conceptual" address.

### PR DESCRIPTION
**Pull request by aarshkshah1992**
For #730 AND Closes #556.

@anorth To review after  #912  goes in.

### CODE REVIEW

**Comment by wadeAlexC** [comment]
LGTM

File: actors/builtin/verifreg/verified_registry_actor.go [hunk 1]
```diff
@@ -52,17 +52,20 @@ type AddVerifierParams struct {
...
 	var st State
 	rt.State().Readonly(&st)
 	rt.ValidateImmediateCallerIs(st.RootKey)
 
+	verifierIdAddr, err := builtin.ResolveToIDAddr(rt, params.Address)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve verifier address %v to ID address", params.Address)
+
+	if params.Allowance.LessThan(MinVerifiedDealSize) {
+		rt.Abortf(exitcode.ErrIllegalArgument, "Allowance %d below MinVerifiedDealSize for add verifier %v", params.Allowance, verifierIdAddr)
+	}
```
Comment (anorth) [inline actors/builtin/verifreg/verified_registry_actor.go:64]: Leave the checks that don't require state above the state load call.

File: actors/builtin/verifreg/verified_registry_actor.go [hunk 2]
```diff
@@ -94,11 +97,14 @@ func (a Actor) RemoveVerifier(rt vmr.Runtime, verifierAddr *addr.Address) *adt.E
 	rt.State().Readonly(&st)
 	rt.ValidateImmediateCallerIs(st.RootKey)
 
+	verifierIdAddr, err := builtin.ResolveToIDAddr(rt, *verifierAddr)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve verifier address %v to ID address", *verifierAddr)
```
Comment (anorth) [inline actors/builtin/verifreg/verified_registry_actor.go:101]: Check before loading state

File: actors/builtin/verifreg/verified_registry_actor.go [hunk 3]
```diff
@@ -114,16 +120,17 @@ type AddVerifiedClientParams struct {
 }
 
 func (a Actor) AddVerifiedClient(rt vmr.Runtime, params *AddVerifiedClientParams) *adt.EmptyValue {
+	vClientIdAddr, err := builtin.ResolveToIDAddr(rt, params.Address)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve verified client address %v", params.Address)
```
Comment (anorth) [inline actors/builtin/verifreg/verified_registry_actor.go:124]: Move after the simple checks, but before the state load

File: actors/builtin/verifreg/verified_registry_actor.go [hunk 4]
```diff
@@ -242,15 +252,16 @@ type RestoreBytesParams struct {
 func (a Actor) RestoreBytes(rt vmr.Runtime, params *RestoreBytesParams) *adt.EmptyValue {
 	rt.ValidateImmediateCallerIs(builtin.StorageMarketActorAddr)
 
+	vClientIdAddr, err := builtin.ResolveToIDAddr(rt, params.Address)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to resolve verified client addr %v", params.Address)
+
 	if params.DealSize.LessThan(MinVerifiedDealSize) {
```
Comment (anorth) [inline actors/builtin/verifreg/verified_registry_actor.go:258]: Check this first

File: actors/builtin/verifreg/verified_registry_actor.go (hunk 1 above)
Comment (aarshkshah1992) [reply actors/builtin/verifreg/verified_registry_actor.go]: Done.

File: actors/builtin/verifreg/verified_registry_actor.go (hunk 2 above)
Comment (aarshkshah1992) [reply actors/builtin/verifreg/verified_registry_actor.go]: Done.

File: actors/builtin/verifreg/verified_registry_actor.go (hunk 3 above)
Comment (aarshkshah1992) [reply actors/builtin/verifreg/verified_registry_actor.go]: Done.

File: actors/builtin/verifreg/verified_registry_actor.go (hunk 4 above)
Comment (aarshkshah1992) [reply actors/builtin/verifreg/verified_registry_actor.go]: Done.

**Comment by aarshkshah1992** [comment]
@anorth Have made all the changes. Please take a look.

File: actors/builtin/verifreg/verified_registry_test.go [hunk 5]
```diff
@@ -184,6 +211,21 @@ func TestRemoveVerifier(t *testing.T) {
 
 		ac.removeVerifier(rt, va)
 	})
+
+	t.Run("add verifier with non ID address and then remove with it's ID address", func(t *testing.T) {
```
Comment (anorth) [inline actors/builtin/verifreg/verified_registry_test.go:215]: Nit: "its"

Comment (aarshkshah1992) [reply actors/builtin/verifreg/verified_registry_test.go]: This is done.

**Pull request merged**